### ISSUE TITLE
GitHub Codespaces setup fails on Jetbrains Gateway® due to file upload error

### ISSUE THREAD

**Issue by redochka** [github/codespaces-jetbrains-feedback#13]
It shows the following error: `Could not connect to Codespace: SSH configuration ID  not found. null`

![image](https://user-images.githubusercontent.com/263297/201306287-c7416d57-d550-4a1c-98e2-7262e85a20e8.png)

The full stack trace is:

```
com.jetbrains.gateway.ssh.deploy.DeployException: Worker binary deploy failed: check command log for details: File /home/codespace/.cache/JetBrains/RemoteDev/remote-dev-worker/remote-dev-worker_8be48ed1f7d9d149a663842f0df2735b7433265d1654774596e54ce70af850f2.1591556543808867.tmp upload failed; stderr: 
	at com.jetbrains.gateway.ssh.GoWorkerDeployer.deploy$intellij_gateway_core(GoWorkerDeployer.kt:216)
	at com.jetbrains.gateway.ssh.GoWorkerDeployer$deploy$1.invokeSuspend(GoWorkerDeployer.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at com.github.codespaces.jetbrains.gateway.connector.CodespacesConnectorService.getJetBrainsConnectionConfig(CodespacesConnectorService.kt:260)
	at com.github.codespaces.jetbrains.gateway.connector.CodespacesConnectorService.access$getJetBrainsConnectionConfig(CodespacesConnectorService.kt:50)
	at com.github.codespaces.jetbrains.gateway.connector.CodespacesConnectorService$prepareForJetBrainsConnection$1.run(CodespacesConnectorService.kt:233)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$8(CoreProgressManager.java:526)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:84)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.jetbrains.gateway.ssh.deploy.DeployException: File /home/codespace/.cache/JetBrains/RemoteDev/remote-dev-worker/remote-dev-worker_8be48ed1f7d9d149a663842f0df2735b7433265d1654774596e54ce70af850f2.1591556543808867.tmp upload failed; stderr: 
	at com.jetbrains.gateway.ssh.deploy.impl.SshPipeFileAccessor.uploadFileFromLocalStream(SshPipeFileAccessor.kt:57)
	at com.jetbrains.gateway.ssh.deploy.impl.SshPipeFileAccessor$uploadFileFromLocalStream$1.invokeSuspend(SshPipeFileAccessor.kt)
	... 32 more
Caused by: java.lang.IllegalStateException: File upload failed with exit code null
	at com.jetbrains.gateway.ssh.deploy.impl.SshPipeFileAccessor.uploadFileFromLocalStream(SshPipeFileAccessor.kt:53)
	... 33 more
```

When ssh on codespace, the file `/home/codespace/.cache/JetBrains/RemoteDev/remote-dev-worker/remote-dev-worker_8be48ed1f7d9d149a663842f0df2735b7433265d1654774596e54ce70af850f2.1591556543808867.tmp` exists however the exception above says it can not upload file

**Comment by ajunior** [comment]
Any news about this? I got the same error.

**Comment by gesellix** [comment]
Seems to work with the latest update.

**Comment by cmuto09** [comment]
Thanks @gesellix- I will close as fixed.

For anyone finding this later, this should be resolved as of the Codespaces plugin version `0.3.0.967`
